### DISCUSSION TITLE
28 - Notifications

### DISCUSSION THREAD

**Discussion by SeriousBug** [dendronhq/dendron#1330] [RFCs/Ideas]
This is the discussion for [RFC 28 - Notifications](https://docs.dendron.so/notes/o5AQmw9bsPQspYuTjdmnK/).

**Comment by SR--** [comment]
Sounds very useful. Our group would definitely be interested. Ideally, this should be integrated with a TODO system, an existing plugin such as Todo Tree (this allows @ mentions, what will happen with those?) or the native one that has been previously mooted.

It will be important to hide or strip those for publishing.

**Reply by ScriptAutomate** [reply]
It would be excellent to see this notification system potentially expanded for tasks / TODO items to not just be for notifying `@user`, but also as reminders associated with dates assigned to notes. I do not know if this is out-of-scope here, but it sounds like the RFC is meant for a notification system, while using `users.*` is then listed an example of using the notification system. [Task notes](https://wiki.dendron.so/notes/SEASewZSteDK7ry1AshNG/) should definitely be included as being able to use the notification system as a way of relaying reminders or alerts on due or upcoming tasks.

**Comment by djradon** [comment]
I think this is a useful and even important feature to have. Confluence has this and it gets used all the time. Even though not always necessary, I think supporting automated email and chat notifications would greatly enhance the value. Could the frontmatter on the user's "user._username_" page include notifying email addresses/chat handles? And maybe notification preferences too?  

@SR-- I can definitely see use cases where people would want to retain these "notifying mentions" in the published version, so hiding/stripping should be optional. Seems like a hard problem though. To take @SeriousBug 's example of "I'm stuck on this issue, can [ ]@Frances take a look at this?" -- would the whole sentence get stripped, or just the "notifying mention"?

**Comment by SR--** [comment]
@djradon Yes, good point. This has been discussed [before](https://github.com/dendronhq/dendron/issues/1178). I agree that this should be an option. And perhaps @ mentions should work within and outside of comments...

Also agree or supporting optional additional means of notifying people, maybe in due course...

**Reply by djradon** [reply]
I can also see it being useful to be able to @-mention and notify groups. Confluence has this functionality.
